android/src/main/kotlin/in/juspay/hyper_sdk_flutter/HyperSdkFlutterPlugin.kt

```diff
@@ -122,16 +122,18 @@ class HyperSdkFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
 
     private fun preFetch(params: Map<String, Any>, result: Result) {
         try {
-            HyperServices.preFetch(binding!!.activity, JSONObject(params))
+            binding?.let { HyperServices.preFetch(it.activity, JSONObject(params)) }
             result.success(true)
         } catch (e: Exception) {
             result.error("HYPERSDKFLUTTER: prefetch error", e.message, e)
         }
     }
 
     private fun initiate(params: Map<String, Any>, result: Result) = try {
-
-        val fragmentActivity = binding!!.activity as FragmentActivity
+        if (binding == null) {
+            result.error("INIT_ERROR", "ActivityPluginBinding is null, cannot proceed", "")
+        }
+        val fragmentActivity = binding?.activity as FragmentActivity
```

Comment:
this should be as follows
```suggestion
        val fragmentActivity = binding?.activity as? FragmentActivity
```